railties/lib/rails/generators/rails/app/templates/config/puma.rb

```diff
@@ -0,0 +1,9 @@
+threads_count = ENV.fetch("MAX_THREADS") { 5 }.to_i
```

Comment:
At the very least, this ENV needs to be prefixed. So either it's PUMA_MAX_THREADS or it's RAILS_MAX_THREADS.